scripts/west_commands/runners/stm32cubeprogrammer.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2020 Teslabs Engineering S.L.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Runner for flashing with STM32CubeProgrammer CLI, the official programming
+   utility from ST Microelectronics.
+"""
+
+from pathlib import Path
+import os
+import shlex
+import sys
+from typing import TYPE_CHECKING, List, Optional, ClassVar, Dict
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+if TYPE_CHECKING:
+    import argparse
+
+    from runners.core import RunnerConfig
+
+
+class STM32CubeProgrammerBinaryRunner(ZephyrBinaryRunner):
+    """Runner front-end for STM32CubeProgrammer CLI."""
+
+    _RESET_MODES: ClassVar[Dict[str, str]] = {
+        "sw": "SWrst",
+        "hw": "HWrst",
+        "core": "Crst",
+    }
+    """Reset mode argument mappings."""
+
+    def __init__(
+        self,
+        cfg: "RunnerConfig",
+        port: str,
+        frequency: Optional[int],
+        reset_mode: Optional[str],
+        extra_connect_opts: Optional[str],
+        prefix: Optional[Path],
+        use_elf: bool,
+        erase: bool,
+        extra_opts: List[str],
+    ) -> None:
+        super().__init__(cfg)
+
+        self._port = port
+        self._frequency = frequency
+        self._reset_mode = reset_mode
+        self._extra_connect_opts = extra_connect_opts
+        self._prefix = prefix
+        self._use_elf = use_elf
+        self._erase = erase
+
+        self._extra_opts: List[str] = list()
+        for opts in [shlex.split(opt) for opt in extra_opts]:
+            self._extra_opts += opts
+
+        self._stm32_programmer_cli = self._get_stm32cubeprogrammer_path()
+
+        # add required library loader path to the environment (Linux only)
+        if sys.platform == "linux":
+            os.environ["LD_LIBRARY_PATH"] = str(
+                self._stm32_programmer_cli.parent / ".." / "lib"
+            )
+
+    def _get_stm32cubeprogrammer_path(self) -> Path:
+        """Find the STM32CubeProgrammer within known paths."""
+
+        if self._prefix:
+            prefix = self._prefix
+        elif sys.platform == "linux":
+            prefix = Path.home()
+        elif sys.platform == "win32":
+            prefix = Path(os.environ["PROGRAMFILES(X86)"])
```

Comment:
My cubeProgrammer instalation in windows is there:
c:\Program Files\STMicroelectronics\STM32Cube\STM32CubeProgrammer\
it is not in c:\Program Files (x86)\

Reply:
Interesting, mine is at x86 folder. Can you determine what is the expected path? Maybe new versions install on non-x86? Are you on a 64-bit system?

Reply:
Yes, I am on 64-bit system. 
I am not sure to be able to determine THE expected path, it looks like there are severalpaths possible (depending on version installed or system version). So I would propose to check the presence of executable in one path and if not available check the other path.

Reply:
Could you list all the *official* paths? In case the user decides to install to a custom location the `--cli` option is available.